source/extensions/quic_listeners/quiche/envoy_quic_proof_verifier_base.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "absl/strings/str_cat.h"
+
+#pragma GCC diagnostic push
+
+// QUICHE allows unused parameters.
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+
+#include "quiche/quic/core/crypto/proof_verifier.h"
+#include "quiche/quic/core/quic_versions.h"
+
+#pragma GCC diagnostic pop
+
+namespace Envoy {
+namespace Quic {
+
+// A partial implementation of quic::ProofVerifier which does signature
+// verification using SSL_SIGN_RSA_PSS_RSAE_SHA256.
+class EnvoyQuicProofVerifierBase : public quic::ProofVerifier {
```

Comment:
I don't understand why EnvoyQuicProofVerifierBase and EnvoyQuicProofVerifier are separate classes.

Reply:
Cert chain verification part is missing in base. In EnvoyQuicProofVerifier  it gets root cert from envoy config.

Reply:
I see that that part is missing in the base class. Why is it missing from the base class instead of combining the two classes into one?